src/NUnitFramework/framework/TestContext.cs

```diff
@@ -142,79 +142,12 @@ public string TestDirectory
 
         #region Static Methods
 
-        /// <summary>Write the string representation of a boolean value to the current result</summary>
-        public static void Write(bool value)
-        {
-            Out.Write(value);
-        }
-
-        /// <summary>Write a char to the current result</summary>
-        public static void Write(char value)
-        {
-            Out.Write(value);
-        }
-
-        /// <summary>Write a char array to the current result</summary>
-        public static void Write(char[]? value)
-        {
-            Out.Write(value);
-        }
-
-        /// <summary>Write the string representation of a double to the current result</summary>
-        public static void Write(double value)
-        {
-            Out.Write(value);
-        }
-
-        /// <summary>Write the string representation of an Int32 value to the current result</summary>
-        public static void Write(int value)
-        {
-            Out.Write(value);
-        }
-
-        /// <summary>Write the string representation of an Int64 value to the current result</summary>
-        public static void Write(long value)
-        {
-            Out.Write(value);
-        }
-
-        /// <summary>Write the string representation of a decimal value to the current result</summary>
-        public static void Write(decimal value)
-        {
-            Out.Write(value);
-        }
-
-        /// <summary>Write the string representation of an object to the current result</summary>
-        public static void Write(object? value)
-        {
-            Out.Write(value);
-        }
-
-        /// <summary>Write the string representation of a Single value to the current result</summary>
-        public static void Write(float value)
-        {
-            Out.Write(value);
-        }
-
-        /// <summary>Write a string to the current result</summary>
-        public static void Write(string? value)
-        {
-            Out.Write(value);
-        }
-
-        /// <summary>Write the string representation of a UInt32 value to the current result</summary>
-        [CLSCompliant(false)]
-        public static void Write(uint value)
-        {
-            Out.Write(value);
-        }
-
-        /// <summary>Write the string representation of a UInt64 value to the current result</summary>
-        [CLSCompliant(false)]
-        public static void Write(ulong value)
-        {
-            Out.Write(value);
-        }
+        /// <summary>
+        /// Write the string representation of an object to the current result
+        /// </summary>
+        /// <param name="val"></param>
+        /// <typeparam name="T"></typeparam>
+        public static void Write<T>(T val) => Out.Write(val);
```

Comment:
This does not what you intended it to do. This always calls `TextWriter.Write(object?)`.
It won't call different methods depending on `T`.
The original code had separate methods per type to prevent boxing.

Reply:
Yes, it does, but it eliminates tons of lines of pretty much useless code but at the same time allows people to use these shortcut methods.  And performance issues should not be an issue when writing to the console, should it?

We have two console output streams (TextWriter), Out and Error.  These methods is just convenience methods for the Out stream.

The methods hides the fact that these go to the Out.  Not sure if that is good or actually bad.

Imho, I would have just killed all of these methods, but that would be a real breaking change.

PS:  If you look at the usage of these methods,  none are used in the Test parts of NUnit.  All usage goes to the string methods.  
Looking at the two most used libs, FluentAssertions and Specflow.Nunit,  none of them seems to use this either.

My goal is not to accomplish extraordinary feats, but to eliminate bloated code

But if you guys feel strongly about it, we'll just leave it as is.

Reply:
I agree with you that it simplifies code and that performance is not really an issue here.

I also would say that they all could go so users explicitly need to use `Out` directly.

What we could do is make an analyzer rule for this with a CodeFix that inserts `.Out` between it. 
Once that is done we can Obsolete these methods and remove them. Technically the latter needs to be done at a major version.

Reply:
Agree that is a good approach!  We need to create some issues for that then.